### PR TITLE
Refactor the form renderer in edit mode

### PR DESCRIPTION
**Pull request by angel-araya-jd**
Components in this refactor: 

>  Decimal Field
>  SingleLine
>  Text Area
>  Float
>  Email
>  Phone
>  BoolLine
>  RadioButton
>  Integer
>  ColorPicker
>  CleanUp

### CODE REVIEW

File: src/lib/components/FormModel/children/Form/UI/Email/index.js [hunk 1]
```diff
@@ -1,3 +1,4 @@
-import Email from './email.component';
+import { Email } from './email.component';
 
+export { Email };
 export default Email;
```
Comment (james-martin-jd) [inline src/lib/components/FormModel/children/Form/UI/Email/index.js]: Extra default export not in other components, remove it for consistency

File: src/lib/components/FormModel/children/Form/UI/Integer/index.js [hunk 2]
```diff
@@ -1,3 +1,4 @@
 import Integer from './integer.component';
 
 export default Integer;
```
Comment (james-martin-jd) [inline src/lib/components/FormModel/children/Form/UI/Integer/index.js]: Extra export not found in other components, let's consolidate for consistency

File: src/lib/components/FormModel/children/Form/UI/Phone/index.js [hunk 3]
```diff
@@ -1,3 +1,4 @@
-import Phone from './phone.component';
+import { Phone } from './phone.component';
 
+export { Phone };
 export default Phone;
```
Comment (james-martin-jd) [inline src/lib/components/FormModel/children/Form/UI/Phone/index.js]: extra export

File: src/lib/components/FormModel/children/Form/UI/RadioButton/index.js [hunk 4]
```diff
@@ -1,3 +1,4 @@
 import RadioButton from './radio-button.component';
 
 export default RadioButton;
+export { RadioButton };
```
Comment (james-martin-jd) [inline src/lib/components/FormModel/children/Form/UI/RadioButton/index.js]: Extra export

File: src/lib/components/FormModel/children/Form/UI/component-mapping.js [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+  [VOCAB.UI.DateField]: DateTimePicker,
+  [VOCAB.UI.DateTimeField]: DateTimePicker,
+  [VOCAB.UI.TimeField]: DateTimePicker,
+  [VOCAB.UI.BooleanField]: CheckBox,
+  [VOCAB.UI.DecimalField]: Decimal,
+  [VOCAB.UI.MultiLineTextField]: TextArea,
+  [VOCAB.UI.FloatField]: Float,
+  [VOCAB.UI.EmailField]: Email,
+  [VOCAB.UI.PhoneField]: Phone,
+  [VOCAB.UI.TriStateField]: RadioButton,
```
Comment (james-martin-jd) [inline src/lib/components/FormModel/children/Form/UI/component-mapping.js]: As has been pointed out, tristate (which is not implemented) is pointing to RadioButton. We can simply remove this item for now until we buiild tristate fields.

File: src/lib/components/FormModel/children/Viewer/UI/BoolLine/index.js [hunk 6]
```diff
@@ -1 +1,4 @@
-export { default } from './bool-line.component';
+import { BoolLine } from './bool-line.component';
+
+export default BoolLine;
+export { BoolLine };
```
Comment (james-martin-jd) [inline src/lib/components/FormModel/children/Viewer/UI/BoolLine/index.js]: Extra export

File: src/lib/components/FormModel/children/Viewer/UI/ui-mapping.js [hunk 7]
```diff
@@ -1,13 +1,11 @@
...
 import { UITypes } from '@constants';
 
-import SingleLine from './SingleLine';
+import {SingleLine} from './SingleLine';
 import MultiLine from './MultiLine';
 import DateLine from './DateLine';
-import BoolLine from './BoolLine';
+import { BoolLine } from './BoolLine';
 import ColorLine from './ColorLine';
-import Heading from '../../Form/UI/Heading';
```
Comment (james-martin-jd) [inline src/lib/components/FormModel/children/Viewer/UI/ui-mapping.js]: Are Heading/Comment no longer in the Viewer, or is it generated another way?

File: src/lib/components/FormModel/form-reducers.js [hunk 8]
```diff
@@ -0,0 +1,11 @@
+export const ModelActions = {
```
Comment (james-martin-jd) [inline src/lib/components/FormModel/form-reducers.js]: I believe this file should be deleted

File: src/lib/components/FormModel/index.js [hunk 9]
```diff
@@ -1,5 +1,3 @@
-import FormModel from './live-form-model.component';
-import AutoSaveDefaultSpinner from './children/autosave-default-spinner.component';
+import { FormModel } from './form-model.component';
```
Comment (james-martin-jd) [inline src/lib/components/FormModel/index.js]: Why was the AutoSaveSpinner deleted from here? It was not strictly part of the form model, but more of an example of how to make autosave spinners (which are dynamically added to the form)

File: src/lib/components/index.js [hunk 10]
```diff
@@ -7,7 +7,7 @@ import ProfileUploader from './ProfileUploader';
 import ShexForm from './ShexForm';
 import ShexFormBuilder from './ShexFormBuilder';
 import ProfileViewer from './ProfileViewer';
-import FormModel, { AutoSaveDefaultSpinner } from './FormModel';
+import FormModel from './FormModel';
```
Comment (james-martin-jd) [inline src/lib/components/index.js]: Why was Autosavedefaultspinner removed?

File: src/lib/index.js [hunk 11]
```diff
@@ -16,7 +16,6 @@ import {
   ShexForm,
   ShexFormBuilder,
   FormModel,
-  AutoSaveDefaultSpinner,
```
Comment (james-martin-jd) [inline src/lib/index.js]: Same question as above, why was this removed?

File: src/lib/components/FormModel/index.js (hunk 9 above)
Comment (angel-araya-jd) [reply src/lib/components/FormModel/index.js]: Re-added a spinner (which now that I think it should be more of a generic indicator) with a caveat. The previous implementation only worked if `autosave` was passed down to the form, if not then there was no indicator at all. Now we have a single indicator for the whole form so it is not next to each field but at the beginning.

It is possible to improve this by updating `formActions.saveData()` to report back which fields were successful and which were not.

Comment (james-martin-jd) [reply src/lib/components/FormModel/index.js]: A couple of notes we discussed offline about the autosavespinner.

First, there are two different save workflows, the first is a full form save, and the second workflow is autosave. This autosave component is intended to only show when autosave is enabled on the form, and is a component per-field that executes on field save. The example default component that ships with the component library is purposely typography-free and not intended for use (I believe it shows text of "Saving..." and "success" or some such). It is intended to be passed in from the client application and rendered that way, as demonstrated in the generator application.

File: src/lib/components/FormModel/children/Viewer/UI/ui-mapping.js (hunk 7 above)
Comment (james-martin-jd) [reply src/lib/components/FormModel/children/Viewer/UI/ui-mapping.js]: I think when we spoke yesterday we determined this file isn't actually used anymore?
